core-data/reducer.js

```diff
@@ -63,48 +68,77 @@ export function media( state = {}, action ) {
 }
 
 /**
- * Reducer managing post types state. Keyed by slug.
+ * Reducer managing theme supports data.
  *
  * @param {Object} state  Current state.
  * @param {Object} action Dispatched action.
  *
  * @return {Object} Updated state.
  */
-export function postTypes( state = {}, action ) {
+export function themeSupports( state = {}, action ) {
 	switch ( action.type ) {
-		case 'RECEIVE_POST_TYPES':
+		case 'RECEIVE_THEME_SUPPORTS':
 			return {
 				...state,
-				...keyBy( action.postTypes, 'slug' ),
+				...action.themeSupports,
 			};
 	}
 
 	return state;
 }
 
 /**
- * Reducer managing theme supports data.
+ * Higher Order Reducer for a given model config. It supports:
  *
- * @param {Object} state  Current state.
- * @param {Object} action Dispatched action.
+ *  - Fetching a record by primariy key
  *
- * @return {Object} Updated state.
+ * @param {Object} modelConfig  Model config.
+ *
+ * @return {Function} Reducer.
  */
-export function themeSupports( state = {}, action ) {
-	switch ( action.type ) {
-		case 'RECEIVE_THEME_SUPPORTS':
-			return {
-				...state,
-				...action.themeSupports,
-			};
-	}
+function model( modelConfig ) {
+	return ( state = { byPrimaryKey: {} }, action ) => {
```

Comment:
What other properties do you foresee living in this state object?

Reply:
I'm thinking `queries` maybe, but it could be added later, I can update this PR to remove this key at the moment.